cogito/putter.go

```diff
@@ -333,6 +333,12 @@ func concourseBuildURL(env Environment) string {
 	// https://ci.example.com/teams/main/pipelines/cogito/jobs/autocat/builds/3?vars=%7B%22branch%22%3A%22stable%22%7D
 	if env.BuildPipelineInstanceVars != "" {
 		buildURL += fmt.Sprintf("?vars=%s", url.QueryEscape(env.BuildPipelineInstanceVars))
+		// Concourse requires that the space characters
+		// are encoded as %20 instead of +. On the other hand
+		// golangs url.QueryEscape as well as url.Values.Encode()
+		// both encode the space as a + character.
+		// See: https://github.com/golang/go/issues/4013
+		buildURL = strings.ReplaceAll(buildURL, "+", "%20")
```

Comment:
If we have a + anywhere else e.g in branch name: "sta+ble" the + would first be escaped with %2B and only remaining + in URL would be the one to escape the space. So this should be safe enough.

Reply:
Sigh. I followed all the links from that issue and I have to agree there isn't any hope. It is also beyond my understanding how Concourse, written in Go, does not accept `+` in the query parameters part.

Thanks for having attempted to do something better.